package/src/custom/online_application_form/ApplyOnline.html

```diff
@@ -0,0 +1,330 @@
+        </div>
+
+        <input id="campaign_id" type="hidden" name="campaign_id" value="will be replaced in submit_form()" />
+        <input id="redirect_url" type="hidden" name="redirect_url" value="will be replaced in submit_form()" />
+        <input id="assigned_user_id" type="hidden" name="assigned_user_id" value="1" />
+        <input id="team_id" type="hidden" name="team_id" value="1" />
+        <input id="team_set_id" type="hidden" name="team_set_id" value="1" />
+        <input id="req_id" type="hidden" name="req_id" value="last_name;" />
+    </form>
+    <p>
+    <script type="text/javascript">// <![CDATA[
+        function submit_form(){
+            // If you were implementing this yourself, you would know the URL of your Sugar instance and could hardcode
+            // it here.  To make this example easy for anyone to try, we are hosting this form on the
+            // Sugar instance (you could host this form anywhere) and pulling the URL of the Sugar instance
+            // from this webpage.
+            var url = window.location.href;
+            var urlBase = url.substring(0, url.lastIndexOf('/custom/online_application_form/ApplyOnline.html'));
+
+            // If you were implementing this yourself, you would hardcode the Campaign ID associated with this form.
+            // To make this example easy for anyone to try, we are getting the Campaign ID dynamically
+            // from Sugar using a custom endpoint.  When the GET request returns with the Campaign ID for the most
+            // recently updated campaign with the name 'Online Applications', we will set the form action, campaign id,
+            // and redirect url.
+            var xhr = new XMLHttpRequest();
+            xhr.withCredentials = true;
+            xhr.addEventListener("readystatechange", function () {
+                if (this.readyState === 4) {
+                    if(xhr.responseText) {
+                        document.getElementById('WebToLeadForm').action = urlBase + "//index.php?entryPoint=WebToLeadCapture";
+
+                        document.getElementById('campaign_id').value = xhr.responseText;
+
+                        document.getElementById('redirect_url').value = urlBase + "/custom/online_application_form/ApplyOnlineSuccess.html";
+
+                        if (typeof(validateCaptchaAndSubmit) != 'undefined') {
+                            validateCaptchaAndSubmit();
+                        } else {
+                            check_webtolead_fields();
+                        }
+                    } else {
+                        alert('We were unable to process your request, because we could not find a Campaign in Sugar with the name Online Applications.');
+                    }
+                }
+            });
+            // Get the ID of the Online Applications Campaign
+            xhr.open("GET", urlBase + "/index.php?entryPoint=getOnlineApplicationsCampaign");
```

Comment:
Why is this an entrypoint? This is simply returning a record ID. If it's meant to be machine consumable then it should be a REST API endpoint. You can make it a public endpoint which should allow it to be accessible without an OAuth token. It should be relatively straight forward to convert to a public rest API.

Reply:
I currently recommend using Thorn to test REST API endpoints, but it may be a bit of work to get it set up.

PHPUnit is certainly adequate until then.